### ISSUE TITLE
Expose XmlFragment and XmlText(Text) blocks

### ISSUE THREAD

**Issue by elwerene** [y-crdt/y-crdt#180]
I'm trying to build a server for tiptap editor with yrs.
I used yrs, forked it and exposed XmlFragment, this worked without too much of a problem.

But the styles of tiptap is not stored in the xml tags, but rather in the linked list of text blocks.
I was able to print out the blocks by adding some functions to XmlText.

There's no public api yet to access these blocks. Do you plan on creating/exposing such an api?

Would you accept a merge request, and if yes, what whould be your requirements for this api?

**Comment by Horusiath** [comment]
> I'm trying to build a server for tiptap editor with yrs.

What actions should this server support? If it's only about aggregation and propagation of peer updates, then current Yrs should work without any changes necessary.

Re. exposing blocks: putting them in public API is not in plans, as they are important part of system internals, and we would like to maintain the freedom to change these in any way necessary once a good reason for this appears. Making them public puts constrains on how internals should look like, while for users they should remain implementation detail.

**Comment by elwerene** [comment]
I would like to provide a document over yrs to clients which is stored in a database as markdown. This document should also be converted back to markdown to save it back after edits. 
Maybe I just didn't see how I can do it otherwise, but it seems I don't have access to text attributes like `bold` and `italic` in another way?
y.js provides access to this, so I thought y.rs should provide it as well?

**Comment by Horusiath** [comment]
I think [Text::diff](https://docs.rs/yrs/0.11.2/yrs/types/text/struct.Text.html#method.diff) method is what you're after, it's basically an equivalent to Yjs `text.toDelta` method.

**Comment by elwerene** [comment]
How would I access Text if I have a XmlText?
And this would only fix the second part of my constraints as I also want to generate a Text object from markdown.

**Comment by elwerene** [comment]
Is it a goal to have the same functionality a y.js? Then we would need a api. I would be willing to write it, but then we would need to discuss how you would like to have it :)

**Comment by Horusiath** [comment]
I'll add remaining methods so that `XmlText` API matches `Text`. Right now there's a possibility to map between one and another, eg.:

```rust
let Doc = Doc::new();
let mut txn = doc.transact();

let xml = txn.get_xml_text("text"); 
let txt = txn.get_text("text"); // same entity but not interpreted as Text
```

It's because underneath, all data types work on the same common `Branch` type, just interpret it differently and expose different APIs to it.

**Comment by elwerene** [comment]
@Horusiath would that help to get/set the attributes as they are used in y-tiptap?

**Comment by Horusiath** [comment]
XmlFragment and XmlText string method changes should be done by now and released as part of v0.13. Feel free to reopen the issue of create a new one if there problem remains.
